integration-tests/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var specificTest string
+var skipBuild bool
+
+func main() {
+	rootCmd := &cobra.Command{
+		Use:   "integration-tests",
+		Short: "Run integration tests",
+		Run:   runIntegrationTests,
+	}
+
+	rootCmd.PersistentFlags().StringVar(&specificTest, "test", "", "Specific test directory to run")
+	rootCmd.PersistentFlags().BoolVar(&skipBuild, "skip-build", false, "Skip building the agent")
+
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func runIntegrationTests(cmd *cobra.Command, args []string) {
+	if !skipBuild {
+		buildAgent()
+	}
+	setupEnvironment()
+
+	if specificTest != "" {
+		fmt.Println("Running", specificTest)
+		if !filepath.IsAbs(specificTest) && !strings.HasPrefix(specificTest, "./tests/") {
+			specificTest = "./tests/" + specificTest
+		}
+		logChan = make(chan TestLog, 1)
+		runSingleTest(specificTest)
+	} else {
+		testDirs, err := filepath.Glob("./tests/*")
+		if err != nil {
+			panic(err)
+		}
+		logChan = make(chan TestLog, len(testDirs))
+		runAllTests()
+	}
+
+	cleanUpEnvironment()
```

Comment:
As a general practice, I think it's better to run this in a defer statement so in case of panics it can be executed. Or when someone adds a new exit point from the function.